source/extensions/common/async_files/async_file_manager_thread_pool.cc

```diff
@@ -0,0 +1,255 @@
+#include "source/extensions/common/async_files/async_file_manager_thread_pool.h"
+
+#include <memory>
+#include <queue>
+#include <thread>
+#include <utility>
+#include <vector>
+
+#include "source/extensions/common/async_files/async_file_action.h"
+#include "source/extensions/common/async_files/async_file_context_thread_pool.h"
+#include "source/extensions/common/async_files/status_after_file_error.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace AsyncFiles {
+
+namespace {
+// ThreadNextAction is per worker thread; if enqueue is called from a callback
+// the action goes directly into ThreadNextAction, otherwise it goes into the
+// queue and is eventually pulled out into ThreadNextAction by a worker thread.
+static thread_local std::shared_ptr<AsyncFileAction> ThreadNextAction;
```

Comment:
`static` can be omitted here as the variable is in an anonymous namespace already.

Reply:
Done.